### ISSUE TITLE
Addon:Docs Zoom in/zoom out impacts docs view

### ISSUE THREAD

**Issue by CodeByAlex** [storybookjs/storybook#8389]
**Describe the bug**
Zooming in and out impacts the look and feel of the docs view
![zoom-in-out-bug](https://user-images.githubusercontent.com/29763590/66662278-89645380-ec16-11e9-8104-e85df8c4b7b8.gif)

**Expected behavior**
Zooming in and out in the canvas should only impact the canvas

**Additional context**
version: 5.3.0-alpha.17
Example found in: Official Storybook - https://monorepo-git-next.storybook.now.sh

**Comment by liamross** [comment]
WORKAROUND: I have a workaround that I implemented to prevent this, it's a script inside of `.storybook/manager-head.html`.

```html
<script>
  const STORYBOOK_IFRAME_SELECTOR = '#storybook-preview-iframe';
  const DOCS_ROOT_SELECTOR = '#docs-root';

  function withIframe(iframe) {
    const iframeWindow = iframe.contentWindow;

    const zoomObserver = new MutationObserver(() => {
      const iframeBody = iframeWindow.document.body;
      const zoom = Number(iframeBody.style.zoom || 1);
      const adjustedZoom = 1 / zoom;
      const docsRoot = iframeBody.querySelector(DOCS_ROOT_SELECTOR);
      if (docsRoot) docsRoot.style.zoom = adjustedZoom;
    });

    iframeWindow.addEventListener('DOMContentLoaded', () => {
      zoomObserver.observe(iframeWindow.document.body, {
        attributes: true,
        childList: true,
        subtree: true,
        attributeFilter: ['style'],
      });
    });
  }

  const nodeAddedObserver = new MutationObserver(mutations => {
    const iframe = document.querySelector(STORYBOOK_IFRAME_SELECTOR);
    if (!iframe || !withIframe) return;
    nodeAddedObserver.disconnect();
    withIframe(iframe);
    withIframe = null;
  });

  window.addEventListener('DOMContentLoaded', () => {
    nodeAddedObserver.observe(document.body, {
      childList: true,
      subtree: true,
      attributes: false,
      characterData: false,
    });
  });
</script>
```

Basically it just divides 1 by the current zoom of the body (the `style.zoom` property of body is what is changing when you zoom in and out) in order to "counter out" the zoom. It works pretty well for me.

**Comment by flynnham** [comment]
This is what I've been using:

```jsx
// preview.js
export const decorators = [
	// your existing decorations...
	function fixZoomEffectForDocs(StoryFn) {
		// Get page params from top level frame (host window)
		const path = (window.top.location.href.split('?')[1] || '')
			.replace('path=', '')

		useEffect(() => {
			if (path.startsWith('/docs')) {
				// Reset zoom to allow normal view
				document.body.style.zoom = 1;
			}
		}, [path]);

		return <StoryFn />;
	},
];
```

**Comment by liamross** [comment]
^ That's definitely a lot simpler than mine, I'll give that a try

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0, and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
